### PR TITLE
Rename MaybeApply to RandomApply

### PR DESCRIPTION
**Pull request by avinair**
There is an issue in the keras_cv repository to rename the MaybeApply layer to RandomApply in order to match standard naming conventions both within and external to keras_cv.
This PR updates the guides in the keras-io repo that uses the MaybeApply layer to use the new name for that layer.

Here are some of the relevant links:
- [keras_cv issue](https://github.com/keras-team/keras-cv/issues/1609)
- [keras_cv PR](https://github.com/keras-team/keras-cv/pull/1792)

Note:
The updated guides haven't been tested since I believe we need to wait for a new release of keras_cv with the change before I can test this.

## Who can review?
@ianstenbit @LukeWood @jbischof

### CODE REVIEW

**Comment by avinair** [comment]
I don't know if there are labels to let people know this PR shouldn't be merged until the new keras_cv release is cut and this PR is tested with the new keras_cv release. If there are, could someone please add it to this PR?

**Comment by jbischof** [comment]
@ianstenbit this guide doesn't seem to be published on keras.io anyway. You can't even find it through Google search. If this is the only use case there's no harm updating before a release.

**Review by jbischof — APPROVED** [review]

**Comment by jbischof** [comment]
Ah, in that case we need to wait @fchollet. We'll push out v0.5.1 in the new few weeks

**Comment by ianstenbit** [comment]
> Thanks for the PR.
> 
> The `RandomApply` does not seem to exist in the latest stable KerasCV release -- @ianstenbit do you know what we should do here?

My plan was that we'd review this PR and only merge it when we cut 0.5.1 (see my comment [here](https://github.com/keras-team/keras-cv/pull/1792#pullrequestreview-1429510416)

**Review by MarkDaoust — APPROVED** [review]

**Comment by MarkDaoust** [comment]
We're past the keras-cv 0.5.1 release, let's merge this.
